imap/search_xapian.c

```diff
@@ -587,199 +614,116 @@ static int tierexists_cb(void *rock, const char *key, size_t keylen,
 
 struct cachetier_rock {
     struct buf *buf;
-    struct db *dst_db;
-    struct txn *txn;
+    struct indexeddb *idb;
 };
 
-static int cachetier_cb(void *rock, const char *key, size_t keylen,
+static int cachetier_cb(void *vrock, const char *key, size_t keylen,
                         const char *data, size_t datalen)
 {
-    /* Ignore all but mailbox entries */
-    if (keylen < 3 || strncmp(key, "*M*", 3)) return 0;
+    struct cachetier_rock *rock = vrock;
+    struct indexeddb *idb = rock->idb;
+
+    if (idb->version >= 2) {
+        /* Ignore all but mailbox entries */
+        if (keylen < 3 || strncmp(key, "*M*", 3)) return 0;
+    }
+    else {
+        /* Ignore cache entries */
+        if (*key == '#') return 0;
+    }
 
-    struct cachetier_rock *mr = rock;
-    size_t prefix_len = buf_len(mr->buf);
+    size_t prefix_len = buf_len(rock->buf);
     int r = 0;
 
-    buf_appendmap(mr->buf, key, keylen);
-    r = cyrusdb_store(mr->dst_db, buf_base(mr->buf), buf_len(mr->buf),
-                      data, datalen, &mr->txn);
-    buf_truncate(mr->buf, prefix_len);
+    buf_appendmap(rock->buf, key, keylen);
+    r = cyrusdb_store(idb->db, buf_base(rock->buf), buf_len(rock->buf),
+            data, datalen, &idb->txn);
+    buf_truncate(rock->buf, prefix_len);
     if (r) {
         syslog(LOG_ERR, "cachetier_cb: could not save key %.*s for tier %s: %s",
-                (int) keylen, key, buf_cstring(mr->buf), cyrusdb_strerror(r));
+                (int) keylen, key, buf_cstring(rock->buf), cyrusdb_strerror(r));
```

Comment:
heh - this syslog still has the same bug that the previous version had - it's printing the key buffer AFTER it truncates it back to just the prefix.  Not a regression, but still wrong.

Reply:
right! fixed that